### ISSUE TITLE
Quantitative results on SemanticKITTI dataset

### ISSUE THREAD

**Issue by stephan000** [ganlumomo/BKISemanticMapping#1]
Hi, @ganlumomo

How are the quantitative results of the SemanticKITTI dataset obtained?

I used the semantickitti_04 data and evaluation code in this repository.
I modified `config/semantickitti.yml` to use all scans:
```
scan_num: 271
``` 

The results of executing the evaluation code are as follows:
```
[ 0.  1.  4.  5.  6.  9. 10. 11. 12. 13. 14. 15. 16. 17. 18. 19.]
[0.00000000e+00 4.33485335e-03 0.00000000e+00 0.00000000e+00
 0.00000000e+00 7.63473164e-04 0.00000000e+00 1.14741928e-06
 8.34710744e-03 1.71361104e-06 9.96031937e-06 1.43424767e-04
 9.64897045e-06 1.47085490e-03 0.00000000e+00 0.00000000e+00]
```

Looking at the estimated semantic map, I see that many points are classified as unlabeled (class 0).
I suspect the parameter settings, but I don't know how to adjust them.

Do you know what's going on?

**Comment by ganlumomo** [comment]
Hi, @stephan000
Thank you for your interest in this project. Class 0 means unoccupied.
I wonder when you visualize the estimated semantic map, do you see any occupied areas?
If so, there might be something incorrect in the evaluation code. I will look into it and let you know.

**Comment by stephan000** [comment]
Thank you for your prompt reply!

The estimated semantic map seems to be somewhat better qualitatively.
![semantickitti_04](https://user-images.githubusercontent.com/53164978/129462979-fc109eef-f295-40fb-94f3-03accd86b5ef.png)

However, most estimated classes are unoccupied.
[000000.txt](https://github.com/ganlumomo/BKISemanticMapping/files/6987334/000000.txt)

The occupied threshold is 0.7, which is the default value in `config/method/semantic_bki.yaml`.
So, Ι think this value is fine.

**Comment by ganlumomo** [comment]
Hi @stephan000,

As the qualitative result looks fine to me. I don't think there's an issue with the code.
The quantitative result given in the paper is obtained for all the semanticKITTI sequences, so I am not sure if the result you got is correct or not.
Also, the 0.7 threshold is not used in this method, the unoccupied voxel is determined by the estimated Categorical distribution.

**Comment by stephan000** [comment]
Hi @ganlumomo ,

To my understanding, the quantitative results listed in the README show the IoU values per scene.
Therefore, I expected that running semantickitti_04, which you provided, would give me the S-BKI results for sequence 04.
Is this incorrect?

It is enough for me to check if semantickitti_04 gives me the right result. However, the results I am getting now seem to be much poorer than the results in the paper.

My environment can be reproduced by docker. If there is a mistake in the procedure, please let me know.
```
FROM ros:kinetic

ENV DEBIAN_FRONTEND=noninteractive

RUN apt update -y -qq && \
  apt upgrade -y -qq --no-install-recommends && \
  apt install -y -qq \
    wget \
    curl \
    unzip \
    ros-kinetic-tf-conversions \
    ros-kinetic-vision-opencv \
    ros-kinetic-pcl-ros \
    ros-kinetic-rviz

ENV CATKIN_WS=/workspace/catkin_ws
WORKDIR ${CATKIN_WS}/src

RUN git clone https://github.com/ganlumomo/BKISemanticMapping && \
  cd ${CATKIN_WS} && \
  . /opt/ros/kinetic/setup.sh && \
  /opt/ros/kinetic/bin/catkin_make 

ARG FILE_ID=19Dv1jQqf-VGKS2qvbygFlUzQoSvu17E5
ARG FILE_NAME=semantickitti_04.zip

RUN cd ${CATKIN_WS}/src/BKISemanticMapping/data && \
  curl -sc /tmp/cookie "https://drive.google.com/uc?export=download&id=${FILE_ID}" > /dev/null && \
  CODE="$(awk '/_warning_/ {print $NF}' /tmp/cookie)" && \
  curl -Lb /tmp/cookie "https://drive.google.com/uc?export=download&confirm=${CODE}&id=${FILE_ID}" -o ${FILE_NAME} && \
  unzip ${FILE_NAME} && \
  rm ${FILE_NAME}

RUN set -x && \
  cd ${CATKIN_WS}/src/BKISemanticMapping/ && \
  sed -i -e "s/^scan_num: 100/scan_num: 271/g" config/datasets/semantickitti.yaml && \
  curl https://bootstrap.pypa.io/pip/3.5/get-pip.py -o get-pip.py && \
  python3 get-pip.py && \
  rm get-pip.py && \
  python3 -m pip install jupyter numpy scikit-learn && \
  jupyter nbconvert --to python semantickitti_evaluation.ipynb && \
  sed -i -e "s/^evaluation_folder.*/evaluation_folder\ =\ '.\/data\/semantickitti_04\/evaluations\/'/g" semantickitti_evaluation.py
```
Please run the following commands:
```
docker build -t bkisemanticmapping .
docker run -dit --name bki_sem_map bkisemanticmapping
docker exec -it bki_sem_map bash
source /workspace/catkin_ws/devel/setup.bash
roslaunch semantic_bki semantickitti_node.launch
cd /workspace/catkin_ws/src/BKISemanticMapping
python3 semantickitti_evaluation.py
```

**Comment by ganlumomo** [comment]
Hi @stephan000 
I just fixed the bug in the commit: [1e18b1e](https://github.com/ganlumomo/BKISemanticMapping/commit/1e18b1e38ff44279dfe8a29f80fe6dbe99ca9449). Apparently, I added a fixed transformation to the map for a better visualization view, but somehow forgot to add the same transformation in the evaluation code.
Sorry for the inconvenience. Also, the results in the paper are obtained using resolution of 0.1. So you might need to change the resolution here: [BKISemanticMapping/config/datasets/semantickitti.yaml](https://github.com/ganlumomo/BKISemanticMapping/blob/master/config/datasets/semantickitti.yaml#L4).
